tests/test_recipes.py

```diff
@@ -16,19 +16,38 @@ def test_recipe_registration(basicApp):
 def test_recipe_build(basicApp, tmpdir):
     basicApp.plugins.get("RecipePlugin")
     test_recipe = basicApp.recipes.get("test_recipe")
+    output_folder = str(tmpdir.mkdir("output"))
+    output_data = test_recipe.build(output_dir=output_folder, no_input=True, extra_context=None)
+    assert output_data == os.path.join(output_folder, "My Package")
+
+
+def test_recipe_build_missing_extra_context_parameter(basicApp, tmpdir):
+    basicApp.plugins.get("RecipePlugin")
+    test_recipe = basicApp.recipes.get("test_recipe")
 
     output_folder = str(tmpdir.mkdir("output"))
-    output_data = test_recipe.build(output_folder, no_input=True)
+    output_data = test_recipe.build(output_dir=output_folder, no_input=True)
 
     assert output_data == os.path.join(output_folder, "My Package")
 
 
+def test_recipe_build_extra_context_parameter_userdata(basicApp, tmpdir):
+    basicApp.plugins.get("RecipePlugin")
+    test_recipe = basicApp.recipes.get("test_recipe")
+
+    output_folder = str(tmpdir.mkdir("output"))
+    user_data = {'project_name': 'TestProject'}
+    output_data = test_recipe.build(output_dir=output_folder, no_input=True, extra_context=user_data)
+
+    assert output_data == os.path.join(output_folder, "TestProject")
```

Comment:
Please check in the test case, if the extra_context content was really used during build() of recipe.

Reply:
Yes, project name created was TestProject which is supplied by user_data.
Otherwise the project name would have been My Package which is mentioned in cookiecutter.json